src/main/java/seedu/financeit/goaltracker/GoalTracker.java

```diff
@@ -0,0 +1,120 @@
+package seedu.financeit.goaltracker;
+
+import seedu.financeit.common.CommandPacket;
+import seedu.financeit.common.Constants;
+import seedu.financeit.common.Goal;
+import seedu.financeit.ui.UiManager;
+import java.util.Scanner;
+
+public class GoalTracker {
+    private static CommandPacket cmdPacket;
+    private static Goal goalSetter = new Goal();
+    private static Scanner scanner = new Scanner(System.in);
+    private static int expenseGoal = 0;
+    private static int incomeGoal = 0;
+    private static String input;
+    private static boolean backToMenu = false;
+
+    public static void main(){
+        boolean exitTracker = false;
+        UiManager.printWithStatusIcon(Constants.PrintType.SYS_MSG, "Welcome to Goals Tracker");
+        System.out.println("[1]. Set goals for Expenses");
+        System.out.println("[2]. Set goals for Income");
+        UiManager.printWithStatusIcon(Constants.PrintType.INSTRUCTION, "Input either expense/income " +
+                "to set goals for it."
+        );
+        cmdPacket = UiManager.handleInput();
+        while(!exitTracker){
+            switch (cmdPacket.getCommandString()){
+                case "expense":
+                    setExpenseGoal();
+                    break;
+                case "income":
+                    setIncomeGoal();
+                    break;
+                case "exit":
+                    exitTracker = true;
+                    break;
+                default:
+                    System.out.println("Invalid Command");
+                    cmdPacket = UiManager.handleInput();
+                    break;
+            }
+        }
+
+    }
+
+    private static void setExpenseGoal(){
```

Comment:
I think the reason why I was not able to exit your sub program is because your event loops here are slightly convoluted and possibly buggy. Please review!